### DISCUSSION TITLE
can I launch an .msixbundle from UWP ?

### DISCUSSION THREAD

**Discussion by jtorjo** [microsoft/WindowsAppSDK#915] [Q&A]
`await Launcher.LaunchFileAsync("path/to/some_file.msixbundle");`

Seems to work on my machine. There's nothing in the docs specifying if this is supposed to work or not. I'd like a clear answer, so I don't once again run into issues on my customer's machines.

## Why am I asking this?

Because I want to grab the `.msixbundle`  from my server, and install it when doing an update

## Why not use an .appinstaller and set the `OnLaunch HoursBetweenUpdateChecks="..."` ?

... which would clearly do just what I want...

Glad you asked: it's because of a bug on your **awesome .msix technology** you've introduced about 1 year ago and I know you won't fix it within the next decade. Namely, updates on an .appinstaller file don't work -- not even if you manually try to do them. So yeah, thanks for that (https://techcommunity.microsoft.com/t5/msix-deployment/windows-10-2004-msix-not-updating-please-check-whether-the/m-p/1466701)

**Comment by ghost** [comment]
> and I know you won't fix it within the next decade.

sums up. Noice there

**Comment by Marv51** [comment]
We install via .appinstaller. Sometimes that will keep the app up-to-date. Our app has a custom checker which checks if updates are needed. If appinstaller does not manage to keep the app up-to-date (unfortunately common, as you mentioned😭) we launch the appinstaller update manually:

```csharp
 await Windows.System.Launcher.LaunchUriAsync("https://example.com/app.appinstaller");
Application.Current.Exit();
```

Not super great, as it requires users to click the "Update" button in the AppInstaller window manually, but it works. With the added benefit that if they fix appinstaller, or for some lucky users, the updates still happen automatically.

**Reply by jtorjo** [reply]
Thanks! 
For some strange reason, launching the appinstaller doesn't work for me - it basically runs into the same issue.
Launching the .msixbundle works.

**Reply by Marv51** [reply]
Maybe `LaunchUriAsync` vs. `LaunchFileAsync`?

**Reply by jtorjo** [reply]
Hmm, I'll have to try it. But my issue is that even if I download the appinstaller manually (externally, via my browser) and launch it, I still run into the issue I initially outlined.
But I will try it.
